server/services/DashboardService/index.js

```diff
@@ -0,0 +1,132 @@
+import { createHash } from 'crypto'
+import _ from 'lodash'
+import CollectionsModel from '../../models/CollectionsModel'
+import MetadataModel from '../../models/MetadataModel'
+
+class DashboardService {
+  constructor(dataDb, study, subject, configuration) {
+    this.assessmentKey = 'analysis'
+    this.collection = 'collection'
+    this.configDataMap = new Map()
+    this.configuration = configuration
+    this.consentDate = ''
+    this.consentDateKey = 'Consent Date'
+    this.consentKey = 'Consent'
+    this.db = dataDb
+    this.hex = 'hex'
+    this.matrixData = []
+    this.role = 'metadata'
+    this.salt = 'sha256'
+    this.study = study
+    this.subject = subject
+    this.subjectKey = 'Subject ID'
+    this.updated = ''
+  }
+
+  _assessmentsFromConfig = () =>
+    _.uniqBy(this.configuration, this.assessmentKey).map(({ analysis }) => {
+      const collectionName = this.study + this.subject + analysis
+      return {
+        assessment: analysis,
+        collection: createHash(this.salt)
+          .update(collectionName)
+          .digest(this.hex),
+      }
+    })
+
+  _dataFromAssessments = async () =>
+    await Promise.all(
+      this._assessmentsFromConfig().map(async ({ collection, assessment }) => {
+        const data = await CollectionsModel.index(this.db, collection)
+        this.configDataMap.set(assessment, data)
+      })
+    )
+
+  _calculateDashboardData = () =>
+    this.configuration.forEach(({ analysis, variable, label, ...rest }) => {
+      const assessmentData = this.configDataMap.get(analysis)
+      const variableData = _.reduce(
+        assessmentData,
+        (variableDataArray, dayData) => {
+          if (Object.hasOwn(dayData, variable)) {
+            variableDataArray.push({
+              day: dayData.day,
+              [variable]: dayData[variable],
+            })
+
+            return variableDataArray
+          }
+
+          return variableDataArray
+        },
+        []
+      )
+
+      if (variableData.length) {
+        const max = _.maxBy(variableData, variable)?.[variable] || null
+        const min = _.minBy(variableData, variable)?.[variable] || null
+        const mean = _.meanBy(variableData, variable) || null
+        const stat = [{ max, min, mean }]
+
+        this.matrixData.push({
+          analysis,
+          data: variableData,
+          label,
+          variable,
+          stat,
+          ...rest,
+        })
+      } else {
+        this.matrixData.push({
+          analysis,
+          data: [],
+          label,
+          variable,
+          stat: [],
+          ...rest,
+        })
+      }
+    })
+
+  setConsentDate = async () => {
+    const metadocAttributes = {
+      study: this.study,
+      role: this.role,
+    }
+    const metadocReference = await MetadataModel.findOne(
```

Comment:
presumably the metadata model returns a metadata instance, what do you think about naming this variable `metadata`?

```suggestion
    const metadata = await MetadataModel.findOne(
```

Reply:
[21d246e](https://github.com/AMP-SCZ/dpdash/pull/344/commits/21d246eb6479a99f1ed04a88b2054072eebd2f2d)